### PR TITLE
Add Gulp support

### PR DESCRIPTION
**Pull request by dymecki**
This PR adds support for Gulp task runner. Using Gulp / node.js tools for building frontend assets is a standard practice today. It offers more packages than Gradle and it will be easier to find new frontend developers.

New src/content/frontend directory is made where Gulp with dependencies are installed. Directories /css, /fonts, /js, /scss, /theme are moved into this directory as well.

Scss task is removed from Gradle. Instead Gradle runs Gulp tasks after "preprocessContent" and "bake" tasks.

No changes were made in scss / css files. This PR is only about Gulp support and better directory structure.

Current tasks:
1. build netbeans.css from netbeans.scss
2. make source maps
3. minify netbeans.css file
4. add license header to netbeans.css file
5. copy netbeans.css file to /build/bake/css
6. copy css/font-awesome.min.css to /build/bake/css
7. copy fonts to /build/bake/fonts
8. copy js files to /build/bake/js

### CODE REVIEW

**Comment by Chris2011** [comment]
Helll yeah, thx. Now we are playing in my field 😎.

**Comment by Chris2011** [comment]
I totally agree with @vieiro. This is the more fronted stack so far. For real frontend users w/o using java shit for the frontend build stuff. Ok this sounds more like a bashing, but yeah, sry for that. I'm fine for now to have it mixed as long as we switch from that gradle stuff to a more common way (mustache, gulp, whatever.)


Cheers

Chris

**Comment by neilcsmith-net** [comment]
I also agree with @vieiro There's a serious need to simplify the website stack, and to move to a framework that is easier to maintain, and easier for new people to understand.  There was an opinion early on about dogfooding (eg. Groovy being Apache) which I had issues with then, and I think has proved a hindrance in practice.  Of course, framework and language maybe should take into account skills we have across the project, but even with knowledge of Java, Groovy and SSG's our system takes a ridiculous amount of time to understand.  That's not to say we necessarily throw away all that's there, which Wade did a great job on, if it could be pared down.

TL;DR - in my opinion this needs a much broader conversation and plan.  I haven't done anything with the website for a while now, but would still be interested in being involved in that conversation.  If it happens post-release! :smile:

@Chris2011 maybe keep the bashing out of this - quite happy to bring up how terrible a lot of the JS stuff is too! :wink:

**Comment by Chris2011** [comment]
> @Chris2011 maybe keep the bashing out of this - quite happy to bring up how terrible a lot of the JS stuff is too! 😉

Sure, but this is the normal/common stack, how frontend works ;).

**Comment by neilcsmith-net** [comment]
> Sure, but this is the normal/common stack, how frontend works ;).

Yes, I know, I've spent well over a decade working there as well! :wink:  Normal / common does not necessarily equate to best / most suitable.  Having a stack that more people understand is one criteria we'll have, just not necessarily the only one.

**Comment by Chris2011** [comment]
This stack, like gulp, grunt, webpack or whatever frontend users know (like me every day) is the stack that a lot of people in the frontend world knows. This is it. Not that Java Gradle/Groovy Stack. This is for Java developers who wants to do frontend as well. Frontend developers stack is different from this and the different is defacto standard, like npm, yarn or whatever. Yes it is common and maybe next day there will be another one, but this is the common stack. When I see vaadin or GWT it is only that Java backend developers want to do Frontend except from CSS and JS.

Anyway +1 for the gulp stuff at the end.

**Comment by vieiro** [comment]
Hi,
Moved the discussion to the mailing list ;-).
Thanks,
Antonio

**Comment by vieiro** [comment]
For the record, infra says node/npm are indeed available in the git-websites nodes:

git-websites-node-npm

**Comment by vieiro** [comment]
No changes provided, as requested. Closing.
